### ISSUE TITLE
User-config is not completely evaluated on startup

### ISSUE THREAD

**Issue by psisis** [syl20bnr/spacemacs#15669]
I'm having a hard time understanding this. For some reason I have issues with code in user-config. I understand that running emacs in daemon-mode can hide messages related to errors in user-config, but I have no errors and this issue still happens when starting emacs directly.

The main issue I have is this

```
(define-key (current-global-map) [mouse-4] 'previous-buffer)
(define-key (current-global-map) [mouse-5] 'next-buffer)
```
Mouse-4 will be set correctly, but not mouse-5. When doing SPC f e R, mouse-5 is now bound as well.

I tried making a weird function that calls spacemacs/sync-configuration-layers (SPC feR) only once after startup (by checking a variable otherwise it would loop indefinitely) but it doesn't solve the issue. Only reloading manually works.

**Comment by lebensterben** [comment]
what do you see with

(describe-key [mouse-5])

**Comment by psisis** [comment]
@lebensterben 

Running that prints `#("<mouse-5> at that spot is undefined" 0 9 (face help-key-binding font-lock-face help-key-binding))`. I manually disabled scrolling with alternative buttons in mouse-mode.

But I should clarify, that piece of code is not the only unevaluated one. For example I have this too
```
  (spaceline-toggle-all-the-icons-eyebrowse-workspace-off)
  (spaceline-toggle-all-the-icons-time-off)
  (spaceline-toggle-all-the-icons-bookmark-on)
  (spaceline-toggle-all-the-icons-dedicated-on)
  (spaceline-toggle-all-the-icons-window-number-off)
  (setq spaceline-all-the-icons-slim-render nil)
  (spaceline-compile)
```
And only after `SPC f e R` do I see the changes. But then, some code is in fact evaluated, like setting mouse-4.
For transparency, I will post my user-config.
<details>
  <summary>User-config</summary>

```
  (require 'use-package)

  (add-hook 'clojure-mode-hook #'lsp-deferred)

  (add-hook 'prog-mode-hook 'undo-tree-mode)

  (use-package yasnippet
    :hook (prog-mode . yas-minor-mode))

  (use-package yasnippet-snippets
    :after (yasnippet))

  (require 'helm-files)
  (require 'helm-command)

  (remove-hook 'cider-repl-mode-hook 'vim-empty-lines-mode)

  (menu-bar-mode 1)
  (context-menu-mode 1)

  (defun sgml-delete-tagged-text ()
    "delete text between the tags that contain the current point"
    (interactive)
    (let ((b (point)))
      (sgml-skip-tag-backward 1)
      (when (not (eq b (point)))
        ;; moved somewhere, should be at front of a tag now
        (save-excursion
          (forward-sexp 1)
          (setq b (point)))
        (sgml-skip-tag-forward 1)
        (backward-sexp 1)
        (delete-region b (point)))))

  (set-window-scroll-bars (minibuffer-window) 0 'none)

  (defun update-scroll-bars ()
    (interactive)
    (mapc (lambda (win)
            (set-window-scroll-bars win nil))
          (window-list))
    (set-window-scroll-bars (selected-window) 10 'right))

  (add-hook 'window-configuration-change-hook 'update-scroll-bars)
  (add-hook 'buffer-list-update-hook 'update-scroll-bars)

  (with-eval-after-load 'cider
    (when (eq system-type 'windows-nt)
      (define-advice cider--list-as-lein-artifact (:override (list &optional exclusions))
        "Add missing double quotes around the version string for cmd.exe."
        (shell-quote-argument
         (format "[%s \"%S\"%s]" (car list) (cadr list) (cider--lein-artifact-exclusions exclusions))))))

  (use-package flycheck-posframe
    :ensure t
    :after flycheck
    :config (add-hook 'flycheck-mode-hook #'flycheck-posframe-mode))

  (add-hook 'company-completion-started-hook
            (lambda (&rest _)
              (setq-local lsp-inhibit-lsp-hooks t)
              (lsp--capf-clear-cache))
            nil
            t)

  (defun quteb ()
    (shell-command "c:\Program/ Files\qutebrowser\qutebrowser.exe localhost:9500"))

  (with-eval-after-load "cider"
    (define-key cider-mode-map (kbd "C-c M-a") 'quteb))

  (spacemacs/set-leader-keys "~" 'spaceline-all-the-icons-toggle-slim)

  (evil-set-initial-state 'cider-repl-mode 'insert)

  (defun customkeys ()
    (define-key (current-global-map) [mouse-4] 'previous-buffer)
    (define-key (current-global-map) [mouse-5] 'next-buffer)
    (define-key dired-mode-map [mouse-1] 'dired-find-file))

  (require 'nerd-fonts)

  (use-package all-the-icons
    :config
    ;; Make sure the icon fonts are good to go
    (set-fontset-font t 'unicode (font-spec :family "all-the-icons") nil 'append)
    (set-fontset-font t 'unicode (font-spec :family "file-icons") nil 'append)
    (set-fontset-font t 'unicode (font-spec :family "Material Icons") nil 'append)
    (set-fontset-font t 'unicode (font-spec :family "github-octicons") nil 'append)
    (set-fontset-font t 'unicode (font-spec :family "FontAwesome") nil 'append)
    (set-fontset-font t 'unicode (font-spec :family "Weather Icons") nil 'append))

  (use-package all-the-icons-nerd-fonts
    :load-path "usr/all-the-icons-nerd-fonts"
    :after all-the-icons
    :demand t
    :config) ;(all-the-icons-nerd-fonts-prefer))

  (use-package spaceline-all-the-icons
    :after spaceline
    :config (spaceline-all-the-icons-theme))

  (spaceline-all-the-icons--setup-anzu)            ;; Enable anzu searching

  (spaceline-toggle-all-the-icons-eyebrowse-workspace-off)
  (spaceline-toggle-all-the-icons-time-off)
  (spaceline-toggle-all-the-icons-bookmark-on)
  (spaceline-toggle-all-the-icons-dedicated-on)
  (spaceline-toggle-all-the-icons-window-number-off)
  (setq spaceline-all-the-icons-slim-render nil)
  (spaceline-compile)

  (defun spreload ()
    (if (boundp 'linit)
        (setq linit nil)
        (setq linit 1))
    (when linit
        (do (dotspacemacs/sync-configuration-layers)
            (remove-hook 'after-make-frame-functions 'spreload))))

  (spacemacs/set-default-font dotspacemacs-default-font)

  (setq-default bidi-inhibit-bpa t)
  (setq-default bidi-paragraph-direction 'left-to-right)

  (setq find-program "c:/lib/msys64/usr/bin/find.exe")

  (customkeys)
```
</details>

**Comment by lebensterben** [comment]
Please comment out all irrelevant lines and retry.

**Comment by psisis** [comment]
I've commented out my entire user-init and only have the define-key commands in user-config. [mouse-5] is still not set.

I do have two warnings on startup which I do not know the cause of, I don't believe it's my doing.

`Local variables list is not properly terminated`
`Error loading autoloads: (user-error Local variables entry is missing the suffix)`

Right now I am not enabling any extra packages in user-config. Another warning is `Package ‘all-the-icons’ isn’t installed` which is caused purely by `dotspacemacs-mode-line-theme '(all-the-icons :separator cup :separator-scale 1.8)` even though I even added it to "additional-packages".

<details>
  <summary>Layers list</summary>

```
   '(ruby
     autohotkey
     csv
     html
     rust
     ;; ----------------------------------------------------------------
     ;; Example of useful layers you may want to use right away.
     ;; Uncomment some layer names and press `SPC f e R' (Vim style) or
     ;; `M-m f e R' (Emacs style) to install them.
     ;; ----------------------------------------------------------------
     (auto-completion :variables
                      spacemacs-default-company-backends '(company-files company-capf)
                      auto-completion-enable-sort-by-usage t
                      auto-completion-enable-snippets-in-popup t
                      auto-completion-enable-help-tooltip t)
     ;; better-defaults
     emacs-lisp
     git
     helm
     lsp
     (haskell :variables haskell-completion-backend 'lsp)
     markdown
     ;multiple-cursors
     org
     (shell :variables
            shell-default-height 30
            shell-default-position 'bottom)
     ;; spell-checking
     syntax-checking
     version-control
     treemacs
     theming
     c-c++
     racket
     common-lisp
     clojure
     ;go
     ;python
     pdf
     javascript
     typescript
     java
     (unicode-fonts :variables unicode-fonts-enable-ligatures t)
     (dash :variables
           dash-autoload-common-docsets t
           dash-docs-docset-newpath "d:/Lib/docsets")
     dap
     spacemacs-org
     spacemacs-purpose
     spacemacs-project
     spacemacs-navigation
     spacemacs-modeline
     spacemacs-layouts
     dtrt-indent
     spacemacs-completion
     spacemacs-defaults
     spacemacs-editing
     spacemacs-editing-visual
     colors
     react
     shell-scripts
     windows-scripts
     json
     ibuffer
     ;tabs
     typography
     )
```
</details>

Messages buffer (not in daemon mode)
```
Local variables list is not properly terminated
Loading c:/Users/x/.spacemacs...done
Package ‘all-the-icons’ isn’t installed
Loading c:/Users/x/.emacs.d/recentf...done
Cleaning up the recentf list...done (0 removed)
Wrote c:/Users/x/.emacs.d/recentf
Error loading autoloads: (user-error Local variables entry is missing the suffix)
Unicode Fonts - Mapping Unicode Blocks in fontset-startup ... done
Unicode Fonts - Mapping Overrides in fontset-startup ... done
Starting a server...
Spacemacs is ready.
Loading display-line-numbers (native compiled elisp)...done
Loading term-cursor (native compiled elisp)...done
Loading t-mouse (native compiled elisp)...done
Loading pixel-scroll (native compiled elisp)...done
Loading c:/Users/x/.emacs.d/elpa/29.0/develop/shackle-20211118.1129/shackle.el (source)...done
Loading c:/Users/x/.emacs.d/.cache/recentf...done
Skipping check for new version (reason: dotfile)
Package cl is deprecated
Waiting for git... [2 times]
```

**Comment by lebensterben** [comment]
Just attach your .spacemacs .
(you may need to rename it to .txt)

**Comment by psisis** [comment]
[spacemacs.txt](https://github.com/syl20bnr/spacemacs/files/9224033/spacemacs.txt)

I've removed comments and the long exec-path var

**Comment by psisis** [comment]
`evil-define-key` works.
